py_openda/py_openda/costFunctions/JObjects.py

```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Module containing various classes that are used by the ensemble kalman filter.
+When adding your own classes be sure that they include all the methods given by the
+default classes JModelFactory, JModelInstance, PyTime and JStochObserver.
+Type conversion for method outputs are taken care of, so there is some flexibility in that regard.
+The data type of inputs are not flexible unless the method only gets called within this script,
+so be sure to carefully read the documentation of the default classes.
+
+Created on Tue Nov 20 15:27:05 2018
+
+@author: hegeman
+"""
+
+import os
+from py4j.java_gateway import JavaGateway
+import py_openda.utils.py4j_utils as utils
+
+gateway = JavaGateway()
+
+
+class JModelFactory:
+    """
+    Wrapperclass for using a stochModelFactory from Java.
+    """
+
+    def __init__(self, config, scriptdir):
+        """
+        :param config: dictionary used for configuration.
+        :param scriptdir: location of the main .oda file.
+        """
+        model_input_dir = os.path.join(scriptdir, config.get('workingDirectory'))
+        model_config_xml = config.get('configFile')
+        self.model_factory = None
+        model_object = 'gateway.jvm.'+config.get('@className')
+        exec('self.model_factory =%s()' % model_object)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of exec detected.](https://app.codacy.com/gh/OpenDA-Association/OpenDA/pullRequest?prid=4902598)